### DISCUSSION TITLE
Improve date/time setting

### DISCUSSION THREAD

**Discussion by markusd112** [Stellarium/stellarium#1788] [Ideas]
In the moment it is a little bit diffcult to change date and time with the horizontal control pane. It would be great, if it could be improved a little bit:
- when clicking to year, month, day, hour, minute or second a control should be displayed to change this data field instantly

The result is, that the date/time control window from the vertical control pane is integrated into the horizontal pane to directly change date/time in ONE control.

You can see this behaviour in other astronomy software and it is much easier to use than in stellarium.

**Comment by alex-w** [comment]
Did you read the text in Help window?

**Reply by markusd112** [reply]
I am somehow familiar using Stellarium... What exactly do you mean? I know how to control date/time. My suggestion is to make it a little bit easier to control it...

**Reply by markusd112** [reply]
ok, you mean the keyboard shortcuts to directly control date/time. My suggestion is to implement a GUI control in the horizontal control pane, to change date/time with the mouse cursor...

**Reply by gzotti** [reply]
There is a date DISPLAY in the bottom bar which does not interact with occasional mouse clicks by design (but gives some extra information on mouse-over), and a date/time CONTROL PANEL. The time control elements in the time control panel are best operated with the mouse wheel. I cannot imagine any easier to use interface. 

If you lack a mouse wheel, you can click on the arrows or enter numbers directly.

**Reply by markusd112** [reply]
But why not use the display to give the possibility to control the time/date? Why navigate to another control panel when you want to change the displayed date/time? For me it is a little bit annoying to open the control panel when I am at display panel... I like it, as it is implemented in other software: there you can control the date/time in the moment you see it.... It's only an idea to improve UI of stellarium...

**Reply by gzotti** [reply]
This is the way our GUI designer has decided a long time ago, and millions of users have grown familiar with this interface. You might accidentally click on the sliding date display and change date. How many people would dislike that?

**Reply by markusd112** [reply]
World is changing everytime. And yes, there could be some users who will be irritated, but from my point of view  it would be more intuitive to have the ability to change something that you see in the moment and not have to go into another dialogue window to CONTROL the date/time. This is a little bit circuitous, but not very intuitive, especially for new users (sorry for my poor english ;-))

**Reply by alex-w** [reply]
> World is changing everytime. And yes, there could be some users who will be irritated, but from my point of view it would be more intuitive to have the ability to change something that you see in the moment and not have to go into another dialogue window to CONTROL the date/time. This is a little bit circuitous, but not very intuitive, especially for new users (sorry for my poor english ;-))

I've checked GUI of KStars and Cartes du Ciel (both have similar GUI for chanding the date and time and it's similar to described one) and I tried run simple task: change the year through GUI. In Stellarium you can change it by 2 clicks. In CdC you can change the year by 3 clicks. In KStars - by 3 clicks and entering year. What about changing year and month? Stellarium - 3 clicks; CdC - 6 clicks; KStars - 4 clicks + typing the year in the GUI.

Note: I didn't touched powerful tool to changing date and time in Stellarium by hotkeys and keyboard+mouse

**Reply by markusd112** [reply]
- Redshift 8: clicking on the displayed year/month/day/hour/minute/second opens a track wheel and you can directly modify the values
- Starry Night Pro: clicking on the displayed year/month/day/hour/minute/second sets the cursor into the data field and the user can directly modify the values entering numbers or using the cursor up/down keys
- SkySafari App: clicking on the displayed year/month/day/hour/minute/second sets the cursor into the data field and you can change the values with the neighboured +/- buttons
- Stellarium App: you click on the displayed time and a slider opens that allows you to easily modify it. A second click on the time opens directly the full control panel for date/time
- Stellarium Web: you click on the displayed time and the full control panel for date/time opens

You can see: there are many popular planetarium software that are doing it as I propose it. I don't want to change the existing way in stellarium I want to offer an **additional** way to control day/time or simply to open the existing day/time control window. Why not open the day/time control window when the user clicks on the displayed day/time? Or place an additional button next to the date/time display to open the control window? 

For my point of view it is not intuitive (especially for the non power user) to have to move the mouse to the vertical menu (which is far away and hidden per default) to open the date/time control window, when you are on the date/time display of the horizontal pane....

**Reply by alex-w** [reply]
A text from Help window:
- F5 toggled Date and Time window
- Ctrl+left mouse drag uses for time dragging
- Ctrl+mouse wheel uses for scrolling minutes
- Ctrl+Shift+mouse wheel uses for scrolling hours
- Ctrl+Alt+mouse wheel uses for scrolling days
- Ctrl+Alt+Shift+mouse wheel uses for scrolling years

Why just don't use it?

**Reply by markusd112** [reply]
...because this is not as intuitive as simply visually clicking on the thing you want to change... the newbie tries to click on the displayed date/time to change it and it doesn't work.  Later on he will maybe find the date/time control button on the unvisible vertical pane, that is far away from date/time display and has no visual conjunction to it.
The keyboard shortcuts are for advanced power users to speed up often used functions. But you have to remember all the ctrl alt shift whatever keys you have to press at the same time. 
I don't want to give up the existing things. I want to give additional possibilities to improve usability for all user groups. The power users can go on using keyboard shortcuts. But tho proposed feature will help beginners and other not power users.

**Reply by alex-w** [reply]
Never heard before that GUI in Stellarium is not intuitive.

Sorry, but your proposal is like a "I'm used to Starry Night - please add similar GUI". I really don't understand why newbie will be clicked on information text to change some properties.

**Reply by markusd112** [reply]
No, I have never said or mean "I'm used to Starr Night - please add similar GUI". 
In my list of programs you can see that also Stellarium (Web and mobile apps) uses exactly this GUI concept, but not the old and good Stellarium PC version. 
I am not a power user, nor a beginner, but I have exactly this problem again and again after a while not using Stellarium PC version, because all the other programs are doing it as I said. In the year 2021 many users are using popular planetarium apps on their smartphones, like SkySafari, Stellarium App, Stellarium Plus App, etc. So they know the GUI concept of these apps and all these apps are working as I described and are irritated when trying to use Stellarium PC.
These users have another background as all the Stellarium PC power users, working with it for several decades.

**Reply by alex-w** [reply]
Sorry, but... proposal to changes in desktop application GUI because smartphone has other GUI looks very weird. This is very different devices which having very different input devices and of course they will have different GUI's. In previously century I was user of RedShift and one of irritates factors was... difficult GUI, which was always visible and small area for the sky on standard monitors. Stellarium may persistent show the left and bottom GUI panels (as other desktop planetariums) but IMHO this is not very good idea for end users.

**Reply by markusd112** [reply]
I really don't understand the problem you have, because it needs no change of the GUI the user see. Only the behaviour when a user tries to click on the date/time display would be different, because it should open the day/time dialogue box (one of my proposed solutions). No additional space is used and none of the known behaviour would be changed.
A GUI concept is not only limited to a device type. Sure there are limitations per device (e.g. screen size, available mouse, keyboard, etc.), but a good GUI is intuitive and the way using a software should be nearly identic, irrespective of the used device. And as I wrote: many PC software are doing it as described, not only smartphone apps. You are concocting my arguments to your needs ;-))
Ok, thanks for this discussion. I think we have heard all the arguments of both sides. I can only propose such an improvement, but the software maintainers have to accept it as an improvement. So I cannot do more. Best regards and thanks to all involved for all the good work on stellarium.
